testnet/stacks-node/src/neon_node.rs

```diff
@@ -1383,6 +1402,101 @@ impl InitializedNeonNode {
         true
     }
 
+    fn get_mining_tenure_information(
+        chain_state: &mut StacksChainState,
+        burn_db: &mut SortitionDB,
+        check_burn_block: &BlockSnapshot,
+        miner_address: StacksAddress,
+        mine_tip_ch: &ConsensusHash,
+        mine_tip_bh: &BlockHeaderHash,
+    ) -> Result<MiningTenureInformation, Error> {
+        let stacks_tip_header = StacksChainState::get_anchored_block_header_info(
+            chain_state.db(),
+            &mine_tip_ch,
+            &mine_tip_bh,
+        )
+        .unwrap()
+        .ok_or_else(|| {
+            error!(
+                "Could not mine new tenure, since could not find header for known chain tip.";
+                "tip_consensus_hash" => %mine_tip_ch,
+                "tip_stacks_block_hash" => %mine_tip_bh
+            );
+            Error::HeaderNotFoundForChainTip
+        })?;
+
+        // the stacks block I'm mining off of's burn header hash and vtxindex:
+        let parent_snapshot =
+            SortitionDB::get_block_snapshot_consensus(burn_db.conn(), mine_tip_ch)
+                .expect("Failed to look up block's parent snapshot")
+                .expect("Failed to look up block's parent snapshot");
+
+        let parent_sortition_id = &parent_snapshot.sortition_id;
+        let parent_winning_vtxindex =
+            SortitionDB::get_block_winning_vtxindex(burn_db.conn(), parent_sortition_id)
+                .expect("SortitionDB failure.")
+                .ok_or_else(|| {
+                    error!(
```

Comment:
Is this error reachable without being indicative of database corruption?  The only way I can think it could be reached is if you're mining off of the genesis block, which has no parent.

Reply:
I think that's correct, but this replicates the previous behavior of the miner (which would also just "not mine" rather than panic here).